lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -514,36 +514,70 @@ export default class BitcoinProcessor {
         // console.debug(`Skipping transaction ${transactionIndex}: no output coins.`);
         continue;
       }
-      const outputs = transactions[transactionIndex].vout as Array<any>;
-      for (let outputIndex = 0; outputIndex < outputs.length; outputIndex++) {
-        // grab the scripts
-        const script = outputs[outputIndex].scriptPubKey;
-
-        // console.debug(`Checking transaction ${transactionIndex} output coin ${outputIndex}: ${JSON.stringify(script)}`);
-        // check for returned data for sidetree prefix
-        const hexDataMatches = script.asm.match(/\s*OP_RETURN ([0-9a-fA-F]+)$/);
-        if (!hexDataMatches || hexDataMatches.length === 0) {
-          continue;
-        }
-        const data = Buffer.from(hexDataMatches[1], 'hex').toString();
-        if (data.startsWith(this.sidetreePrefix)) {
-          // we have found a sidetree transaction
-          const sidetreeTransaction: TransactionModel = {
-            transactionNumber: TransactionNumber.construct(block, transactionIndex),
-            transactionTime: block,
-            transactionTimeHash: blockHash,
-            anchorString: data.slice(this.sidetreePrefix.length)
-          };
-          console.debug(`Sidetree transaction found; adding ${JSON.stringify(sidetreeTransaction)}`);
-          await this.transactionStore.addTransaction(sidetreeTransaction);
-          // stop processing future anchor strings. Protocol defines only the first should be accepted.
-          break;
-        }
+
+      try {
+        const outputs = transactions[transactionIndex].vout as Array<any>;
+
+        await this.addSidetreeTransactionsToTransactionStore(this.transactionStore, outputs, transactionIndex, block, blockHash);
```

Comment:
Consider renaming the method `addSidetreeTransactionsToTransactionStore` to something like `processTransactionVOuts` instead. Because the current name seems to imply that the transaction will for sure be a Sidetree transaction.

Reply:
Renaming to addValidSidetreeTransactionsFromVOutsToTransactionStore(…)